Runtime/Core/ApiGatewayManagement/AnywhereGameServerAdapter.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Amazon.GameLift;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement.Models;
+using AmazonGameLiftPlugin.Core.Shared;
+using UnityEngine;
+using Logger = AmazonGameLiftPlugin.Core.Shared.Logging.Logger;
+using Amazon.GameLift.Model;
+
+namespace AmazonGameLiftPlugin.Core.ApiGatewayManagement 
+{
+    public class AnywhereGameServerAdapter : IGameServerAdapter
+    {
+        private readonly IAmazonGameLiftClientWrapper _amazonGameLiftClientWrapper;
+        private readonly string _fleetId;
+        private readonly string _location;
+        private readonly string _playerIdPrefix = "playerId-";
+        
+        public AnywhereGameServerAdapter(IAmazonGameLiftClientWrapper amazonGameLiftClientWrapper, string fleetId, string fleetLocation )
+        {
+            _amazonGameLiftClientWrapper = amazonGameLiftClientWrapper;
+            _location = fleetLocation;
+            _fleetId = fleetId;
+        }
+
+        public async Task<GetGameConnectionResponse> GetGameConnection(GetGameConnectionRequest request)
+        {
+            try
+            {
+                var describeGameSessionsResponse = await _amazonGameLiftClientWrapper.DescribeGameSessions(new DescribeGameSessionsRequest
+                {
+                    FleetId = _fleetId,
+                    StatusFilter = GameSessionStatus.ACTIVE
+                });
+
+                var oldestGameSession = describeGameSessionsResponse.GameSessions.First();
```

Comment:
Should this be moved to inside the if block? Wondering what happens when there are 0 game sessions